engine/runtime/src/main/java/org/enso/interpreter/runtime/data/hash/EnsoHashMapBuilder.java

```diff
@@ -82,6 +88,51 @@ public StorageEntry[] getEntries(int atGeneration, int size) {
     }
   }
 
+  public record Entry(Object key, Object value) {}
```

Comment:
No public methods, classes, etc. in the `hash` package, please. I am not sure if this one is visible or not (because `EnshHashMapBuilder` isn't public), but still, the less `public` the better.